RoboJackets/igvc-software
# Feat/ground truth node cpp class

# Description

This PR converts the ground_truth node into a class with a header and cpp file.

Fixes #739 

# Testing steps (If relevant)
1. Run a simulation on the noetic-migration branch
2. Run the same simulation on the feat/ground_truth_node_cpp_class branch
3. compare the results and make sure they are the same 

Expectation: Both results 

# Self Checklist
- [ ] I have tested that the new behaviour works (Need reviewer to test since VM lags heavily with simulation)
